client/src/index.js

```diff
@@ -1,29 +1,35 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import { Router, Route, Switch } from 'react-router-dom';
+import { history } from './history';
+import { Provider } from 'react-redux';
+import { store } from './redux/store';
+
 import './index.css';
 import reportWebVitals from './reportWebVitals';
-import { App } from './components/App';
-import { CreateList } from './components/CreateList';
-import { ViewList } from './components/ViewList';
-import { Find } from './components/Find';
-import { NewPlace } from './components/NewPlace';
-import { EditPlace } from './components/EditPlace';
+import { connectApp } from './components/App';
+import { connectCreateList } from './components/CreateList';
+import { connectEditPlace } from './components/EditPlace';
+import { connectViewList } from './components/ViewList';
+import { connectFind } from './components/Find';
+import { connectNewPlace } from './components/NewPlace';
 import { Menu } from './components/Menu';
 
 ReactDOM.render(
   <React.StrictMode>
-    <BrowserRouter>
-      <Menu />
-      <Switch>
-        <Route path="/" exact component={App} />
-        <Route path="/lists/new" exact component={CreateList} />
-        <Route path="/lists/:id" exact component={ViewList} />
-        <Route path="/find" exact component={Find} />
-        <Route path="/places/new" exact component={NewPlace} />
-        <Route path="/places/:id" exact component={EditPlace} />
-      </Switch>
-    </BrowserRouter>
+    <Provider store={store}>
+      <Router history={history}>
+        <Menu />
+        <Switch>
+          <Route path="/" exact component={connectApp} />
```

Comment:
All these smart components should be in `containers` or `views` directory, as opposed to dumb/pure components